src/java/main/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -146,14 +146,26 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
             );
             containerManager.start();
 
+            // Watch status of ZooKeeper server. If there is an internal error
+            // then will do a graceful shutdown.
+            while (zkServer.isRunning()) {
```

Comment:
@rakeshadr sure, I understand that an interface would allow us to do a different version, but do we right now envision any other way of implementing this? if not, then I'd just **not** do an interface and know that we can do one if it ever becomes a need. I'd rather avoid a proliferation of unnecessary interfaces, otherwise we will end up doing everything with an interface.